File: JavaLibrary/src/test/java/com/flat/simplex/parser/logic/LineParserTest.java
package com.flat.simplex.parser.logic;

import com.flat.simplex.parser.logic.block.BlockIf;
import com.flat.simplex.parser.logic.line.LineValue;
import com.flat.simplex.parser.logic.line.call.CallField;
import com.flat.simplex.support.LineCallChain;
import com.flat.simplex.support.TokenChain;
import org.junit.jupiter.api.Test;

import static com.flat.simplex.support.ContextSupport.assertErrors;
import static com.flat.simplex.support.LineCallChain.lChain;
import static com.flat.simplex.support.TokenChain.readChain;

class LineParserTest {

    @Test
    void parse() {
        TokenChain chain = readChain("a.b.c");

        Context context = new Context(chain.get());
        BlockIf block = getBlock(context);
        LineValue line = new LineParser(block, chain.get(), null).parse();

        LineCallChain callChain = lChain(CallField.class, CallField.class, CallField.class);
        callChain.assertChain(line, "Invalid line chain");
        assertErrors(context);
    }

    private BlockIf getBlock(Context context) {
        TokenChain chain = readChain("if(true);");

        return new BlockIf(null, chain.get(), null);

    }
}

File: SimplexApp/simplex/parser/logic/LineParser.test.js
const TokenChain = require("simplex/support/TokenChain.js");
const LineCallChain = require("simplex/support/LineCallChain.js");
const LineParser = require("simplex/parser/logic/LineParser.js");
const Context = require("simplex/parser/logic/Context.js");
const BlockIf = require("simplex/parser/logic/block/BlockIf.js");
const ContextSupport = require("simplex/support/ContextSupport.js");

const readChain = TokenChain.readChain;
const lChain = LineCallChain.lChain;
const assertErrors = ContextSupport.assertErrors;

test('parse', () => {
    let chain = readChain("a.b.c");

    let context = new Context(chain.get());
    let block = getBlock(context);
    let line = new LineParser(block, chain.get(), null).parse();

    let callChain = lChain('CallField', 'CallField', 'CallField');
    callChain.assertChain(line, "Invalid line chain");
    assertErrors(context);
});

function getBlock(context) {
    let chain = readChain("if(true);");

    return new BlockIf(context, chain.get(), null);
}